mlflow/tensorflow.py

```diff
@@ -1050,39 +1050,50 @@ def _get_early_stop_callback_attrs(callback):
         except Exception:  # pylint: disable=W0703
             return None
 
-    def _log_early_stop_callback_metrics(callback, history, expected_last_epoch, metrics_logger):
-        if callback:
-            callback_attrs = _get_early_stop_callback_attrs(callback)
-            if callback_attrs is None:
-                return
-            stopped_epoch, restore_best_weights, patience = callback_attrs
-            metrics_logger.record_metrics({"stopped_epoch": stopped_epoch})
-
-            # Only log restored model metrics if early stopping occurs, as determined by the
-            # the value of `stopped_epoch`. `stopped_epoch` is non-zero if early stopping has
-            # occurred, except in the case where training was early stopped after the first epoch
-            # due to a configured patience value of zero
-            if (
-                (stopped_epoch > 0) or (patience == 0 and expected_last_epoch > 0)
-            ) and restore_best_weights:
-                restored_epoch = stopped_epoch - patience
-                metrics_logger.record_metrics({"restored_epoch": restored_epoch})
-                restored_index = history.epoch.index(restored_epoch)
-
-                restored_metrics = {
-                    key: history.history[key][restored_index] for key in history.history.keys()
-                }
-                # Metrics are logged as 'epoch_loss' and 'epoch_acc' in TF 1.X
-                if Version(tensorflow.__version__) < Version("2.0.0"):
-                    if "loss" in restored_metrics:
-                        restored_metrics["epoch_loss"] = restored_metrics.pop("loss")
-                    if "acc" in restored_metrics:
-                        restored_metrics["epoch_acc"] = restored_metrics.pop("acc")
-                # Checking that a metric history exists
-                metric_key = next(iter(history.history), None)
-                if metric_key is not None:
-                    last_epoch = len(history.history[metric_key])
-                    metrics_logger.record_metrics(restored_metrics, last_epoch)
+    def _log_early_stop_callback_metrics(callback, history, metrics_logger):
```

Comment:
`keras.py` and `tensorflow.py` contains several duplicate functions. We might want to move them in e.g. `utils.py`.